src/physical_object.py

```diff
@@ -45,24 +45,80 @@ def get_forces_composition(self):
             force[1] += self.forces[f][1]
         return force
 
+    def reset_forces(self):
```

Comment:
You do that every iteration, otherwise the previous iteration's gravity will affect current gravity.